File: tests/fixtures/sample.ts
/**
 * Sample TypeScript fixture for testing the parser.
 * Contains functions, classes, interfaces, types, and enums.
 */

import { EventEmitter } from 'node:events';
import type { Readable } from 'node:stream';

/** A greeting function */
export function greet(name: string): string {
  return `Hello, ${name}!`;
}

/** An async data fetcher */
export async function fetchData(url: string, timeout?: number): Promise<string> {
  const response = await fetch(url);
  return response.text();
}

export const add = (a: number, b: number): number => a + b;

export interface User {
  id: number;
  name: string;
  email: string;
  isActive: boolean;
}

export type UserId = number | string;

export enum Status {
  Active = 'active',
  Inactive = 'inactive',
  Pending = 'pending',
}

/** A repository class for managing users */
export class UserRepository {
  private users: Map<string, User> = new Map();

  /** Create a new user */
  async createUser(data: Omit<User, 'id'>): Promise<User> {
    const user: User = { ...data, id: Date.now() };
    this.users.set(String(user.id), user);
    return user;
  }

  /** Find a user by ID */
  getUser(id: string): User | undefined {
    return this.users.get(id);
  }

  /** Get all users */
  listUsers(): User[] {
    return [...this.users.values()];
  }

  static fromJSON(json: string): UserRepository {
    const repo = new UserRepository();
    return repo;
  }
}

const API_VERSION = '1.0.0';

File: tests/fixtures/sample.py
"""
Sample Python fixture for testing the parser.
Contains functions, classes, and methods.
"""

import os
from pathlib import Path


def greet(name: str) -> str:
    """Return a greeting string."""
    return f"Hello, {name}!"


async def fetch_data(url: str, timeout: int = 30) -> str:
    """Fetch data from a URL."""
    return ""


class UserRepository:
    """A repository for managing users."""

    def __init__(self, db_path: str):
        """Initialize the repository."""
        self.db_path = db_path
        self._users = {}

    def create_user(self, name: str, email: str) -> dict:
        """Create a new user."""
        user = {"name": name, "email": email}
        self._users[name] = user
        return user

    def get_user(self, name: str) -> dict | None:
        """Get a user by name."""
        return self._users.get(name)

    def _internal_method(self):
        """A protected method."""
        pass

    def __private_method(self):
        """A private method."""
        pass

    @staticmethod
    def from_json(data: str) -> "UserRepository":
        """Create from JSON."""
        return UserRepository("")


class AdminRepository(UserRepository):
    """Extended repository for admins."""

    def list_admins(self) -> list:
        """List all admins."""
        return []


API_VERSION = "1.0.0"
